terraform/README.md

```diff
@@ -0,0 +1,67 @@
+# Deploying with Terraform + Juju + Kubernetes
+
+The `testflinger.tf` plan in this directory can be used for deploying
+the Testflinger server into a dev, staging, or production environment.
+For development, it is recommended to use Juju with Microk8s.
+
+
+## Deploying an Environment
+
+The following assumes that you have an existing Juju + k8s environment
+with a `k8s-certification` credential imported in juju for connecting
+a new model to the k8s environment.
+
+For purposes of this example, we will be deploying a **staging**
+environment.
+
+This also requires that you import the TLS certificate for https into
+a k8s secret. For the example below the name of the secret used was:
+`testflinger-staging-canonical-com-tls`. You will need to do this
+*after* creating the model in step 1.
+
+1. First, create the model
+```
+    $ juju add-model testflinger-staging --credential k8s-certification
+```
+
+2. The terraform juju provider doesn't currently support specifying storage
+requirements. So we need to work around this by manually deploying the
+database. You can modify the specified storage requirements to fit your needs.
+```
+    $ juju deploy -m testflinger-staging --channel=5/edge mongodb-k8s --storage mongodb=10G
+```
+
+3. (optional) Export environment variables to be used in the following steps. If
+you choose not to do this, you'll need to either specify them on the command line
+for each command, or answer the interactive prompts from terraform.
+```
+    $ export TF_VAR_environment=staging
+    $ export TF_VAR_external_ingress_hostname=testflinger-staging.canonical.com
+    $ export TF_VAR_tls_secret_name=testflinger-staging-canonical-com-tls
+```
+
+4. (First run only) Initialize terraform
+```
+    $ terraform init
+```
+
+5. Since we specify the model name (testflinger-${environment}) in the terraform plan,
+and we already created it by hand to deploy mongodb, we need to import that
+existing model into the terraform state so that terraform doesn't give
+us an error when it tries to create it.
+```
+    $ terraform import juju_model.testflinger_model testflinger-staging
+```
+
+6. Running `terraform plan` will show us what it intends to do without
+actually doing it. Look at the output from this and make sure it looks
+correct.
+```
+    $ terraform plan
+```
+
+7. Finally, apply the terraform plan. After this settles, everything should
```

Comment:
Worth also noting after this as a separate section what to do past this initial setup (just `terraform apply` all the way down after that) when you want to make changes to the system.

Reply:
Added a comment about this